syntax = "proto3";

import "EchoAttributes.proto";

package network;
option java_package = "com.philips.emil.protobufEcho";
option java_outer_classname = "NetworkProto";

message IpConfig
{
    bool usesDhcp = 1;
    fixed32 ipAddress = 2;
    fixed32 netmask = 3;
    fixed32 gateway = 4;
}

message MacAddress
{
    bytes address = 1 [(bytes_size) = 6];
}

message PrivateCertificate
{
    string certificate = 1 [(string_size) = 4096];  // The certificate and privateKey must include a terminating 0
    string privateKey = 2 [(string_size) = 2048];
}

message PublicCertificate
{
    bytes certificate = 1 [(bytes_size) = 1536];
}

message WiFiStation
{
    network.IpConfig ipConfig = 1;
    string key = 2 [(string_size) = 64];
    string ssid = 3 [(string_size) = 32];
    bytes bssid = 4 [(bytes_size) = 6];
    uint32 channel = 5;

    enum Security
    {
        open = 0x0;
        wpa2MixedPsk = 0x400006;
        wpa3Psk = 0x800000;
        unknown = -1;
    }

    Security security = 6;
}

message WiFiAccessPoint
{
    string ssid = 1 [(string_size) = 32];
}

message WiFiAccessPointWithPassword
{
    string ssid = 1 [(string_size) = 32];
    string key = 2 [(string_size) = 64];
}

message MbedTlsPersistedSession
{
    bool clientSessionObtained = 1;
    bytes serializedSession = 2 [(bytes_size) = 512];
    bytes identifier = 3 [(bytes_size) = 32];
}
